python/hail/dataset.py

```diff
@@ -2745,6 +2745,52 @@ def linreg_multi_pheno(self, ys, covariates=[], root='va.linreg', use_dosages=Fa
                                            jarray(Env.jvm().java.lang.String, covariates), root, use_dosages, min_ac,
                                            min_af)
         return VariantDataset(self.hc, jvds)
+    
+    @handle_py4j
+    @requireTGenotype
+    @typecheck_method(ys=listof(strlike),
+                      covariates=listof(strlike),
+                      root=strlike,
+                      use_dosages=bool,
+                      variant_block_size=integral)
+    def linreg3(self, ys, covariates=[], root='va.linreg', use_dosages=False, variant_block_size=16):
```

Comment:
You can still use DenseMatrix and slice it after you process the block.  I considered that, but as I've said before I don't think it belongs here.  I'd rather focus on improving general mechanisms to handle this use case.  If you want to submit it afterwards I wouldn't reject it.